packages/desktop-client/src/components/ManageRules.js

```diff
@@ -495,15 +505,82 @@ function RulesList({
   );
 }
 
-export default function ManageRules({
-  isModal,
-  payeeId,
-  setLoading = () => {}
-}) {
+function mapValue(field, value, { payees, categories, accounts }) {
+  if (!value) return '';
+
+  if (field === 'payee') {
+    return payees.find(p => p.id === value).name;
+  } else if (field === 'category') {
+    return categories.find(c => c.id === value).name;
+  } else if (field === 'account') {
+    return accounts.find(a => a.id === value).name;
+  } else {
+    return value;
+  }
+}
+
+function ruleToString(rule, data) {
+  let conditions = rule.conditions.flatMap(cond => [
+    mapField(cond.field),
+    friendlyOp(cond.op),
+    cond.op === 'oneOf'
+      ? cond.value.map(v => mapValue(cond.field, v, data)).join(', ')
+      : mapValue(cond.field, cond.value, data)
+  ]);
+  let actions = rule.actions.flatMap(action => {
+    if (action.op === 'set') {
+      return [
+        friendlyOp(action.op),
+        mapField(action.field),
+        'to',
+        mapValue(action.field, action.value, data)
+      ];
+    } else if (action.op === 'link-schedule') {
+      let schedule = data.schedules.find(s => s.id === action.value);
+      return [
+        friendlyOp(action.op),
+        describeSchedule(
+          schedule,
+          data.payees.find(p => p.id === schedule._payee)
+        )
+      ];
+    } else {
+      return [];
+    }
+  });
+  return (
+    (rule.stage || '') + ' ' + conditions.join(' ') + ' ' + actions.join(' ')
+  );
+}
+
+function ManageRulesContent({ isModal, payeeId, setLoading }) {
   let [allRules, setAllRules] = useState(null);
   let [rules, setRules] = useState(null);
+  let [filter, setFilter] = useState('');
   let dispatch = useDispatch();
   let navigator = useTableNavigator(rules, ['select', 'edit']);
+
+  let { data: schedules } = SchedulesQuery.useQuery();
+  let filterData = useSelector(state => ({
+    payees: state.queries.payees,
+    categories: state.queries.categories.list,
+    accounts: state.queries.accounts,
+    schedules
+  }));
+
+  console.log('fil', filterData);
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏oops